packages/kit/src/runtime/client/client.js

```diff
@@ -1241,30 +1237,54 @@ function get_rerouted_url(url) {
  * @param {URL | undefined} url
  * @param {boolean} invalidating
  */
-function get_navigation_intent(url, invalidating) {
+async function get_navigation_intent(url, invalidating) {
 	if (!url) return;
 	if (is_external_url(url, base, app.hash)) return;
 
-	const rerouted = get_rerouted_url(url);
-	if (!rerouted) return;
-
-	const path = get_url_path(rerouted);
+	if (__SVELTEKIT_CLIENT_ROUTING__) {
+		const rerouted = get_rerouted_url(url);
+		if (!rerouted) return;
+
+		const path = get_url_path(rerouted);
+
+		for (const route of routes) {
+			const params = route.exec(path);
+
+			if (params) {
+				const id = get_page_key(url);
+				/** @type {import('./types.js').NavigationIntent} */
+				const intent = {
+					id,
+					invalidating,
+					route,
+					params: decode_params(params),
+					url
+				};
+				return intent;
+			}
+		}
+	} else {
+		/** @type {{ route?: import('types').CSRRouteServer, params: Record<string, string>}} */
+		const { route, params } = await import(
+			/* @vite-ignore */
+			new URL(
+				base + `/${app.app_dir}/routes` + (url.pathname === '/' ? '.js' : url.pathname + '.js'),
+				location.href
+			).href
```

Comment:
pretty sure we can just do this instead of all the `URL` stuff:

```suggestion
			base + `/${app.app_dir}/routes` + (url.pathname === '/' ? '.js' : url.pathname + '.js')
```

...though `url.pathname` isn't quite right, because it will include any `paths.base`, i.e. if you have this in your config file...

```js
paths: {
  base: '/potato'
}
```

...then if you visit the `/foo` route it will try to load `/potato/_app/routes/potato/foo.js` which is wrong. We need to strip out the basepath first, so I guess it's more like this:

```suggestion
			base + `/${app.app_dir}/routes` + (url.pathname.slice(base.length) === '/' ? '.js' : url.pathname.slice(base.length) + '.js')
```

Side-note, `/_app/routes.js` feels odd for the root route, maybe it should just be `/_app/route` instead of `_app/routes`? Not a big deal either way

Reply:
No, it's correct, because the base path is stripped on the server before this gets to the route resolution. We can simplify it though with the URL a bit